### PR TITLE
{AzureDataShare} fixes Azure/azure-powershell#20822 fix the description

### PR DESCRIPTION
**Pull request by navba-MSFT**
fixes Azure/azure-powershell#20822 fix the description

The StorageAccountResourceId parameter in the "New-AzDataShareDataSetMapping" PowerShell cmdlet is used to specify the resource ID of the Azure storage account that is the source of the data being shared. This parameter is required when creating a new data set mapping for an Azure Data Share and is used to identify the storage account from which the data will be shared. The resource ID of the storage account can be obtained from the Azure portal or through Azure PowerShell.

The AccountName parameter in the "New-AzDataShareDataSetMapping" PowerShell cmdlet is used to specify the name of the Azure storage account that is the source of the data being shared. This parameter is required when creating a new data set mapping for an Azure Data Share and is used to identify the storage account from which the data will be shared. The account name of the storage account must be unique within Azure and can be used to access the storage account and the data stored within it.

The ResourceGroup parameter in the "New-AzDataShareDataSetMapping" PowerShell cmdlet is used to specify the name of the Azure resource group that contains the Azure storage account being shared. This parameter is required when creating a new data set mapping for an Azure Data Share and is used to identify the resource group that holds the storage account from which the data will be shared. The resource group can be used to manage the resources within it, such as the storage account and the data set mapping.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/DataShare/DataShare/help/New-AzDataShareDataSetMapping.md [hunk 1]
```diff
@@ -210,7 +210,7 @@ Accept wildcard characters: False
 ```
 
 ### -StorageAccountResourceId
-Azure Storage Account ResourceId
+Azure Storage Account ResourceId that is the source of the data being shared.
```
Comment (BethanyZhou) [inline src/DataShare/DataShare/help/New-AzDataShareDataSetMapping.md:213]: Hi @navba-MSFT , thanks for updating description for these parameters. Their descriptions in source code need to be updated at the same time. See https://github.com/Azure/azure-powershell/blob/fc2fa77a49e122e7ab22397f9a652d8370711402/src/DataShare/DataShare/DataSetMapping/NewAzDataShareDataSetMapping.cs#L62 and https://github.com/Azure/azure-powershell/blob/fc2fa77a49e122e7ab22397f9a652d8370711402/src/DataShare/DataShare/DataSetMapping/NewAzDataShareDataSetMapping.cs#L107

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
BTW, I think these parameters also exist in Get-AzDataShareDataSetMapping and Remove-AzDataShareDataSetMapping. Please double check whether their descriptions need to be updated as well.

File: src/DataShare/DataShare/help/New-AzDataShareDataSetMapping.md (hunk 1 above)
Comment (navba-MSFT) [reply src/DataShare/DataShare/help/New-AzDataShareDataSetMapping.md:213]: done.

File: src/DataShare/DataShare/DataSetMapping/NewAzDataShareDataSetMapping.cs [hunk 2]
```diff
@@ -104,7 +104,7 @@ public class NewAzDataShareDataSetMapping : AzureDataShareCmdletBase
         [Parameter(
             Mandatory = true,
             ValueFromPipelineByPropertyName = true,
-            HelpMessage = "Azure Storage Account ResourceId",
+            HelpMessage = "Azure Storage Account ResourceId that is the source of the data being shared.",
```
Comment (BethanyZhou) [inline src/DataShare/DataShare/DataSetMapping/NewAzDataShareDataSetMapping.cs:107]: Help message for AccountName should be updated as well. Also, please check Get/Update/Remove/SetAzDataShareDataSetMapping.

Comment (navba-MSFT) [reply src/DataShare/DataShare/DataSetMapping/NewAzDataShareDataSetMapping.cs:107]: done

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**